### PR TITLE
ARROW-12915: [Release] Build of ubuntu-docs is failing on thrift

### PR DESCRIPTION
**Pull request by jonkeane**

### CODE REVIEW

**Comment by jonkeane** [comment]
This PR resolves building on Ubuntu 20.04 [which is our default](https://github.com/apache/arrow/blob/master/.env#L45) though the crossbow build in tasks is [on 20.10](https://github.com/apache/arrow/blob/master/dev/tasks/tasks.yml#L949) which has one of these additions, but also a newer protobuff so that's not needed to be bundled. 

We also might consider _not_ adding these lines but making clear in the docker compose file that `ubuntu-docs` will not work with Ubuntu 20.04 and instead should always be build on 20.10

**Comment by jonkeane** [comment]
I ran this on crossbow (locally since the GH actions comment bot isn't working right now https://issues.apache.org/jira/browse/ARROW-12919) and the only ubuntu crossbow builds that failed were all ones that have other current nightly failures.

I've also added a defaults docs nightly to catch if we have a mismatch like this in the future.

File: ci/docker/ubuntu-20.04-cpp.dockerfile [hunk 1]
```diff
@@ -75,6 +75,7 @@ RUN apt-get update -y -q && \
         libcurl4-openssl-dev \
         libgflags-dev \
         libgoogle-glog-dev \
+        libgrpc++-dev \
```
Comment (kou) [inline ci/docker/ubuntu-20.04-cpp.dockerfile:78]: We don't need to install `libgrpc++-dev` and `protobuf-compiler-grpc` when we use bundled gRPC and ProtoBuf.

**Review by jorgecarleitao — APPROVED** [review]
Thanks a lot, @jonkeane !

**Pull request closed without merging**